sdk/samples/core/mosquitto_rpc_server_sample.c

```diff
@@ -362,32 +383,28 @@ int main(int argc, char* argv[])
   connection_options.client_certificates[0] = primary_credential;
 
   LOG_AND_EXIT_IF_FAILED(az_mqtt5_connection_init(
-      &iot_connection, &connection_context, &mqtt5, iot_callback, &connection_options));
+      &mqtt_connection, &connection_context, &mqtt5, iot_callback, &connection_options));
 
   pending_command.request_data = AZ_SPAN_FROM_BUFFER(request_payload_buffer);
   pending_command.content_type = AZ_SPAN_FROM_BUFFER(content_type_buffer);
   pending_command.correlation_id = AZ_SPAN_EMPTY;
   pending_command.response_topic = AZ_SPAN_FROM_BUFFER(response_topic_buffer);
-  pending_command.command_name = AZ_SPAN_FROM_BUFFER(command_name_buffer);
+  pending_command.request_topic = AZ_SPAN_FROM_BUFFER(request_topic_buffer);
 
   az_mqtt5_property_bag property_bag;
   LOG_AND_EXIT_IF_FAILED(az_mqtt5_property_bag_init(&property_bag, &mqtt5, NULL));
 
-  rpc_server_options
-      = (az_mqtt5_rpc_server_options){ .sub_topic = AZ_SPAN_FROM_BUFFER(sub_topic_buffer),
-                                       .command_name = command_name,
-                                       .model_id = model_id };
-
   az_mqtt5_rpc_server_memory rpc_server_memory
-      = (az_mqtt5_rpc_server_memory){ .property_bag = property_bag };
+      = (az_mqtt5_rpc_server_memory){ .property_bag = property_bag,
```

Comment:
API naming - I would either rename this to rpc_server_context or completely remove it and accept the property bag and subscribe_topic as arguments.

Also, per API design guidelines, the rpc_server_options must exist on all '_init()' functions (constructors). 
- In this case it could be just a struct `_internal { uint8_t reserved }`. 
- A corresponding `az_mqtt5_rpc_server_memory_options_default()` function should also exist. (When we do have options, the application side pattern would be to call this function first, then modify non-internal members before passing the options to the _init function.)

Reply:
Removed the memory struct and moved the property bag and subscribe topic to be arguments on init